packages/models/files.js

```diff
@@ -7,6 +7,12 @@ const Files = new FilesCollection({
     debug: true,
     collectionName: 'files',
     protected(fileObj) {
+        var session = this.params.query.xmtok;
+        var userId  = (Meteor.server.sessions[session] && Meteor.server.sessions[session].userId) ? Meteor.server.sessions[session].userId : null;
```

Comment:
@menelike there was something which decodes `hashedToken` to user's `_id` (*or other persistent field on user's record*)